src/common/providers/EdgeConfig/EdgeConfigProvider.tsx

```diff
@@ -1,33 +1,44 @@
 import { getAll } from "@vercel/edge-config";
 import { edgeConfigSchema } from "@/server/ecfg/config";
-
 import { EdgeConfigContextProvider } from "./EdgeConfigContextProvider";
 
+async function fetchAndValidateEdgeConfig() {
+  try {
+    const rawConfig = await getAll();
+    const result = edgeConfigSchema.safeParse(rawConfig);
+
+    if (!result.success) {
+      console.warn(
+        "Failed to parse edge config:\n",
+        result.error.message,
+        "\nReceived config:\n",
+        rawConfig,
+      );
+      return null;
+    }
+
+    return result.data;
+  } catch (error) {
+    console.warn(
+      "Failed to fetch edge config:\n",
+      error instanceof Error ? error.message : error,
+    );
+    return null;
+  }
+}
+
+async function getFallbackConfig() {
+  return (await import("@/server/ecfg/config.json")).default;
+}
+
 export async function EdgeConfigProvider({
   children,
 }: {
   children: React.ReactNode;
 }) {
-  // * FUTURE NOTE *
-  // if edge requests count nears the threshold,
-  // we should consider caching the edge config & revalidate every 24h
-  const edgeConfigRaw = await getAll();
```

Comment:
could we leave this comment in? :D

it'll be a useful reference for the future if/when we need to do performance optimization for it

Reply:
added back! [8e00293](https://github.com/AfterClass-io/afterclass.io-v2/pull/345/commits/8e00293559b7beb47f5ee3f3d793d8a626b966b0)